src/tests/ftest/util/ssh_connection.py

```diff
@@ -0,0 +1,137 @@
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+from paramiko import client
+
+class Ssh(object):
+    """
+    It's mimic to the subprocess class, only difference is to use ssh protocol
+    and get command execution on remote machine where in subprocess command will
+    be initiated on local machine.
+    Right now check_output() and call() function implemented.In future more
+    methods can be added which will behave like subprocess methods.
+    """
+    def __init__(self, address, timeout=60, debug=False):
+        """
+        Initialize the remote machines and open SSH Connection.
+        Args:
+            address: remote machine IP address or hostname.
+            timeout: Timeout for waiting to finish the command
+            debug  : To print the command on console.
+        return:
+            None
+        """
+        self.address = address
+        self.timeout = timeout
+        self.debug = debug
+        self.client = client.SSHClient()
+        self.client.set_missing_host_key_policy(client.AutoAddPolicy())
+
+    def connect(self):
+        """
+        Connect the client via ssh connection
+        """
+        self.client.connect(self.address)
+
+        if self.debug:
+            print("SSH Connection open to {0}.".format(self.address))
+
+    def check_output(self, command):
+        """
+        Execute command on remote machine.
+        Args:
+            command: Command to run on remote machine.
+        return:
+            stdout: stdout of the return code
+        Raises:
+            general: if it's failed in ssh channel setup, exec command, close()
+            general: If command exit status is not 0.
+        """
+        try:
+            transport = self.client.get_transport()
+            transport.set_keepalive(20)
+            channel = transport.open_session()
+            channel.settimeout(self.timeout)
+            channel.exec_command(command)
+            stdout = channel.makefile('rb').read()
+            exit_status = channel.recv_exit_status()
+            channel.close()
+        except:
+            print("Exception in call for command {0}".format(command))
+            raise
+
+        if self.debug:
+            stderr = channel.makefile_stderr('rb').read()
+            print("[{0}] Command :{1}\nRC:{2} \nstdout : {3}\nstderr :{4}"
+                  .format(self.address,
+                          command,
+                          exit_status,
+                          stdout,
+                          stderr))
+
+        if exit_status != 0:
+            raise Exception("[{0}] Command :{1}\nRC:{2} \nstdout : {3}\n" \
```

Comment:
I'm not sure I agree with this.  What if we want to run a command and do something based it's return status?  What about a function that returns a tuple of return_status, stdout, and stderr and a second function that calls that method and checks the return_code and either raises an exception with stderr or returns stdout?

Reply:
That's a different method.  This class is mimicking `subprocess`:
https://github.com/daos-stack/daos/blob/7e73b36e09d212bdaa51a70fa0a685e3e0fe2ba2/src/tests/ftest/util/ssh_connection.py#L29-L33

Reply:
Sorry, i missed that.  Thanks for the clarification.